ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemAdd.java

```diff
@@ -215,6 +221,35 @@ protected void populateModel(final OperationContext context, ModelNode operation
         }
     }
 
+    /*
+     * Conditional registration of capabilities for default bean instance pools (which may or may not be defined)
+     */
+    @Override
+    protected void recordCapabilitiesAndRequirements(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+        ModelNode model = resource.getModel();
+
+        // register the capability we are exporting as well as its capability requirement on the strict-max-pool that supports it
+        if (model.hasDefined(DEFAULT_SLSB_INSTANCE_POOL)) {
+            context.registerCapability(DEFAULT_SLSB_POOL_CONFIG_CAPABILITY);
+
+            // need to resolve the attribute value before using it
+            String resolvedDefaultSLSBPoolName = context.resolveExpressions(model.get(DEFAULT_SLSB_INSTANCE_POOL)).asString();
```

Comment:
It's unfortunate this attribute allows expressions. This runs in Stage.MODEL and some expressions (typically legacy security vault ones) may not be resolvable until Stage.RUNTIME. Which is the big reason we avoid allow expressions for model reference attributes. But this one has been around forever, so it is what it is.

To be really safe L236-238 should be in a try/catch that just logs a WARN if resolvesExpressions throws OperationFailedException. We'd just not record the requirement.

Reply:
Done